tests/unit/test_sagemaker_steps.py

```diff
@@ -901,6 +901,119 @@ def test_transform_step_creation(pca_transformer):
+                'InstanceType': step_input['instance_type'],
+                'VolumeKmsKeyId': execution_input['volume_kms_key']
+            },
+            'Tags': execution_input['tags'],
+            'Environment': execution_input['env'],
+            'MaxConcurrentTransforms': execution_input['max_concurrent_transforms'],
+            'MaxPayloadInMB': execution_input['max_payload']
+        }
+
+    step = TransformStep('Inference',
+        transformer=pca_transformer,
+        data=execution_input['data'],
+        data_type=execution_input['data_type'],
+        content_type=execution_input['content_type'],
+        compression_type=execution_input['compression_type'],
+        split_type=execution_input['split_type'],
+        job_name=execution_input['job_name'],
+        model_name=execution_input['model_name'],
+        experiment_config={
+            'ExperimentName': 'pca_experiment',
+            'TrialName': 'pca_trial',
+            'TrialComponentDisplayName': 'Transform'
+        },
+        tags=execution_input['tags'],
+        join_source=execution_input['join_source'],
+        output_filter=execution_input['output_filter'],
+        input_filter=execution_input['input_filter'],
+        parameters=parameters
+    )
+
+    assert step.to_dict()['Parameters'] == {
+        'BatchStrategy.$': "$$.Execution.Input['strategy']",
+        'ModelName.$': "$$.Execution.Input['model_name']",
+        'TransformInput': {
+            'CompressionType.$': "$$.Execution.Input['compression_type']",
+            'ContentType.$': "$$.Execution.Input['content_type']",
+            'DataSource': {
+                'S3DataSource': {
+                    'S3DataType.$': "$$.Execution.Input['data_type']",
+                    'S3Uri.$': "$$.Execution.Input['data']"
+                }
+            },
+            'SplitType.$': "$$.Execution.Input['split_type']"
+        },
+        'TransformOutput': {
+            'Accept.$': "$$.Execution.Input['accept']",
+            'AssembleWith.$': "$$.Execution.Input['assemble_with']",
+            'KmsKeyId.$': "$$.Execution.Input['output_kms_key']",
+            'S3OutputPath.$': "$$.Execution.Input['output_path']"
+        },
+        'TransformJobName.$': "$$.Execution.Input['job_name']",
+        'TransformResources': {
+            'InstanceCount.$': "$$.Execution.Input['instance_count']",
+            'InstanceType.$': "$['instance_type']",
+            'VolumeKmsKeyId.$': "$$.Execution.Input['volume_kms_key']"
+        },
+        'ExperimentConfig': {
+            'ExperimentName': 'pca_experiment',
+            'TrialName': 'pca_trial',
+            'TrialComponentDisplayName': 'Transform'
+        },
+        'DataProcessing': {
+            'InputFilter.$': "$$.Execution.Input['input_filter']",
+            'OutputFilter.$': "$$.Execution.Input['output_filter']",
+            'JoinSource.$': "$$.Execution.Input['join_source']",
+        },
```

Comment:
like the variety. how about covering if a type with nested properties is allocated dynamically (i.e. `TransformResources`, `ExperimentConfig`, or `DataProcessing` entirely)